courses/react-js/intermidiate-level/managing-complex-uis/TodoItem.js

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+function TodoItem({ todo, toggleTodo }) {
+  const { id, text, completed } = todo;
+
+  const handleToggle = () => {
+    toggleTodo && toggleTodo(id);
+  };
+ 
+
+  return (
+    <div style={{ marginBottom: "5px" }}>
```

Comment:
**Visible, non-interactive elements with click handlers must have at least one keyboard listener.** 

Resolving for new code

```jsx
    <span
        style={{
          textDecoration: completed ? "line-through" : "none",
          marginLeft: "10px",
          cursor: "pointer",
          color: completed ? "var(--ifm-task-complete-color)" : "var(--ifm-task-incomplete-color)",
        }}
        onClick={handleToggle}
        onKeyDown={handleKeyPress}
        tabIndex={0}
      >
        {text}
    </span>
```